strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/nuget/filter/NugetFilterODataParserTestCase.java

```diff
@@ -76,7 +77,104 @@ public void testSearchConjunction(@NugetRepository(repositoryId = REPOSITORY_REL
+                                                   versions = { "1.0.0",
+                                                                "1.0.1",
+                                                                "1.0.2" })
+                                Path artifactPath)
+    {
+        Selector<ArtifactEntry> selector = new Selector<>(ArtifactEntry.class);
+
+        NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(
+                "tolower(Id) eq 'org.carlspring.strongbox.nuget.test.nfpt_-test' and IsLatestVersion");
+        Predicate predicate = t.parseQuery().getPredicate();
+
+        selector.where(predicate)
+                .and(Predicate.of(ExpOperator.EQ.of("storageId", repository.getStorage().getId())))
+                .and(Predicate.of(ExpOperator.EQ.of("repositoryId", repository.getId())));
+
+        selector.select("count(*)");
+
+        QueryTemplate<Long, ArtifactEntry> queryTemplate = new OQueryTemplate<>(entityManager);
+	
         assertThat(queryTemplate.select(selector)).isEqualTo(1L);
     }
 
+    /*
+     * Tests that starting dash doesn't work
+     */
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    @Transactional
+    public void testSearchCharsStart(@NugetRepository(repositoryId = REPOSITORY_RELEASES) Repository repository,
+                                     @NugetTestArtifact(repositoryId = REPOSITORY_RELEASES,
+                                                        id = "-Org.Carlspring.Strongbox.Nuget.Test.Nfpt",
+                                                        versions = { "1.0.0",
+                                                                     "1.0.1",
+                                                                     "1.0.2" })
+                                     Path artifactPath)
+    {
+        Selector<ArtifactEntry> selector = new Selector<>(ArtifactEntry.class);
+
+        NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(
+                "tolower(Id) eq '-org.carlspring.strongbox.nuget.test.nfpt'");
+        Predicate predicate = t.parseQuery().getPredicate();
+
+        selector.where(predicate)
+                .and(Predicate.of(ExpOperator.EQ.of("storageId", repository.getStorage().getId())))
+                .and(Predicate.of(ExpOperator.EQ.of("repositoryId", repository.getId())));
+
+        selector.select("count(*)");
+
+        QueryTemplate<Long, ArtifactEntry> queryTemplate = new OQueryTemplate<>(entityManager);
+	
+        assertThat(queryTemplate.select(selector)).isEqualTo(0L);
+    }
+
+    /*
+     * Tests that ending dash doesn't work
+     */
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    @Transactional
+    public void testSearchCharsEnd(@NugetRepository(repositoryId = REPOSITORY_RELEASES) Repository repository,
```

Comment:
this case seems no longer needed in connection with this [comment](https://github.com/strongbox/strongbox/pull/1472/files#r336779033)

Reply:
I don't think the nuget layout provider tests are running. They don't put any output to the logs like the other providers and this test is incorrect but the program still compiles

Reply:
@grantfar: Did you try renaming this class to something like `NugetFilterODataParserTest`, like I mentioned earlier? This will probably solve the problem with it not being executed.